### PR TITLE
Update to Angular 5.2 and SDK 2.1

### PR DESCRIPTION
**Pull request by kwalrath**
No real code changes were required, although I did make one small 2.1 change to use the `mixin` keyword (thanks to dartfix). I didn't remove unnecessary imports of dart:async, since there was no easy way to do it globally.

Staged: https://kw-webdev-dartlang-1.firebaseapp.com/changelog

### CODE REVIEW

**Comment by kwalrath** [comment]
@chalin, note that I stopped using variables for version #s. Our current changelog has the wrong version #s, thanks to its use of variables, so I figured we should just copy over what's in the master pubspec. (I got the July 2018 #s from https://github.com/dart-lang/site-webdev/blob/da826866f55da5c3dd5a075a2ce9c61e9d46e0eb/src/_data/pubspec.yaml.)

File: src/angular/note/faq/dependency-injection.md [hunk 1]
```diff
@@ -0,0 +1,315 @@
+---
+title: "Dependency Injection FAQ"
+excerpt_separator: ""
+---
```
Comment (chalin) [inline src/angular/note/faq/dependency-injection.md]: @kwalrath - you'll need to manually delete `src/angular/note/faq/di.md`.

Comment (chalin) [reply src/angular/note/faq/dependency-injection.md]: Deleting this comment since it isn't relevant: git figured out that it was a rename.

**Comment by kwalrath** [comment]
I plan to merge this Monday, followed immediately by dart-lang/stagehand#591 and then the angular-examples repos.

**Comment by kwalrath** [comment]
I thought that changing src/_data/pubspec.yaml would update the examples, but it didn't. I'll need to make another PR.

/cc @chalin

**Comment by kwalrath** [comment]
Or not. @chalin is of the opinion that updating the pubspec.yaml files isn't necessary.

**Pull request merged**